src/StackInterface.php

```diff
@@ -1,11 +1,11 @@
 <?php
 
-namespace Psr\Http\Middleware;
+namespace Idealo\Middleware;
 
-use Psr\Http\Message\RequestInterface;
-use Psr\Http\Message\ResponseInterface;
+use Interop\Http\ServerMiddleware\DelegateInterface;
+use Interop\Http\ServerMiddleware\MiddlewareInterface;
 
-interface StackInterface
+interface StackInterface extends DelegateInterface
 {
     /**
      * Return an instance with the specified middleware added to the stack.
```

Comment:
I totally agree. For adding middlewares on runtime `withMiddleware(MiddlewareInterface $middleware)` does not really do a good job - it just appends middleware. There are better ways like conditional middlewares or simple routing (eg. via a fastroute router) to other stacks.

Let´s drop `StackInterface` and implement `DelegateInterface` and cleanup the withs ...

I will update the PR.